tests/test_ble_interface_core.py

```
@@ -388,29 +453,1263 @@ def test_state_manager_closing_only_for_disconnect() -> None:
+    monkeypatch.setattr(
+        "meshtastic.interfaces.ble.interface.BLEClient",
+        lambda *_args, **_kwargs: pytest.fail(
+            "Temporary BLEClient should not be created when target is owned elsewhere"
+        ),
+    )
+
+    with pytest.raises(BLEInterface.BLEError, match=ERROR_CONNECTION_SUPPRESSED):
+        if method_name == "pair":
+            iface.pair("mesh-node", confirm=True, await_timeout=7.0)
+        else:
+            iface.unpair("mesh-node", await_timeout=7.0)
```

Ruff (0.15.4)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)